extensions/amp-video/0.1/video-cache.js

```diff
@@ -85,17 +88,39 @@ function selectVideoSource(videoEl) {
  */
 function applySourcesToVideo(videoEl, sources, maxBitrate) {
   sources
-    .sort((a, b) => a['bitrate_kbps'] - b['bitrate_kbps'])
+    .sort((a, b) => {
+      // 'codec' values can contain metadata after the '.' that we must strip
+      // for sorting purposes. For example, "vp09.00.30.08" contains level,
+      // profile, and color depth values that are ignored in this sort.
+      const aCodec = a['codec']?.split('.')[0];
+      const bCodec = b['codec']?.split('.')[0];
+
+      // The greater the index, the more the codec is preferred.
+      const aCodecPriority = CODECS_IN_ASCENDING_PRIORITY.indexOf(aCodec);
+      const bCodecPriority = CODECS_IN_ASCENDING_PRIORITY.indexOf(bCodec);
+
+      // A positive difference means that source A has a superior codec.
+      const codecDifference = aCodecPriority - bCodecPriority;
```

Comment:
Can `codec` be nullish? Because of the `?.` optional chain, `aCodec`/`bCodec` can be `undefined`. This leads to a `-1` priority for either, meaning an unknown codec will become the highest priority item.

Reply:
Thanks for the review! I think the existing logic should work because we treat higher indices as having higher priorities here, so unknown codecs with `-1` priorities should actually be treated as being the lowest priority items.

Reply:
Thank you for the review Justin, appreciate it as always :))
It should be covered by [this test](https://github.com/ampproject/amphtml/pull/36224/files#diff-619248237fea1a24fba2b6d01b6dc5b8696ad016640acd4a36f68f20f24344c9R162)

Reply:
Ahh. I understand now. The comparator's comments are incorrect (positive difference means "b before a"), so the sorted array is going from least -> greatest. But, the insertion inserts the new nodes at the beginning of the `<video>` _during each iteration_ (it's mutating essentially).

As nits, this should be fixed, but as long as it's tested, it's okish.